### ISSUE TITLE
Accessibility/usability of popover on hover event

### ISSUE THREAD

**Issue by blackbaud-johnly** [blackbaud/skyux-popovers#91]
### Expected behavior
The popover component meets accessibility requirements and the docs and demo provide correct guidance. 

At a recent SCD meeting about using popovers, a couple people raised concerns about triggering popovers on hover instead of on click posing a potential accessibility issue. I reached out to @Blackbaud-MattGregg to ask about this, and he said it is an accessibility and usability problem to use only hover. We currently offer a first-class option to display on hover and promote it in the demo. Should we really have this option? If so, should we promote it in the demo without any accompanying docs to explain the how to use the option and still meet accessibility requirements? The Design guidelines include an item about using `click` actions rather than `mouseenter` actions to trigger popovers on touch devices, but I'm not sure if this is sufficient guidance. Do we need to change the component or update the docs?
